src/services/pull/useSingularImpactedFileComparison.tsx

```diff
@@ -0,0 +1,225 @@
+import { useQuery } from '@tanstack/react-query'
+import z from 'zod'
+
+import {
+  FirstPullRequestSchema,
+  MissingBaseCommitSchema,
+  MissingBaseReportSchema,
+  MissingComparisonSchema,
+  MissingHeadCommitSchema,
+  MissingHeadReportSchema,
+} from 'services/comparison'
+import {
+  RepoNotFoundErrorSchema,
+  RepoOwnerNotActivatedErrorSchema,
+} from 'services/repo/schemas'
+import Api from 'shared/api'
+import { NetworkErrorObject } from 'shared/api/helpers'
+import A from 'ui/A'
+
+import { FileComparisonWithBase } from './fragments'
+import { transformImpactedFileData } from './utils'
+
+const query = `
+    query ImpactedFileComparison($owner: String!, $repo: String!, $pullId: Int!, $path: String!, $filters: SegmentsFilters) {
+      owner(username: $owner) {
+        repository(name: $repo) {
+          __typename
+          ... on Repository {
+            pull(id: $pullId) {
+              ...FileComparisonWithBase
+            }
+          }
+          ... on NotFoundError {
+            message
+          }
+          ... on OwnerNotActivatedError {
+            message
+          }
+        }
+      }
+    }
+    ${FileComparisonWithBase}
+`
+
+const ImpactedFileSchema = z.object({
```

Comment:
Can we import this from `useComparisonForCommitAndParent` just so we have consistent definitions and clean up the file a bit?

Reply:
I'd rather leave it as it is since they're not using any shared query/fragment. This would cause errors if we ever changed one of these two hooks' schemas. It's a bad idea for different hooks' schemas to depend on each other.